### PR TITLE
feat: allow more customization of the custom homepage

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

This allows users to modify the actual grid properties from backstage interface.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by drodil** [comment]
Not stale, waiting for review

**Comment by drodil** [comment]
Not stale, waiting for review

**Comment by camilaibs** [comment]
Starting to review 🙂

File: plugins/home/src/components/CustomHomepage/CustomHomepageGrid.tsx [hunk 1]
```diff
@@ -316,10 +333,22 @@ export const CustomHomepageGrid = (props: CustomHomepageGridProps) => {
       <ResponsiveGrid
         className={styles.responsiveGrid}
         measureBeforeMount
-        compactType="horizontal"
+        compactType={props.compactType}
```
[the remark this answers is not in the document]
Comment (drodil) [reply plugins/home/src/components/CustomHomepage/CustomHomepageGrid.tsx:402]: Have to postpone the tests once more, calendar is just too tight.

File: plugins/home/src/components/CustomHomepage/CustomHomepageGrid.tsx [hunk 2]
```diff
@@ -180,13 +180,30 @@ const availableWidgetsFilter = (elements: ElementCollection) => {
...
+  containerPadding?:
+    | [number, number]
+    | Record<BreakPointOption, [number, number]>;
+  margin?: [number, number] | Record<BreakPointOption, [number, number]>;
+  maxRows?: number;
+  style?: React.CSSProperties;
+  compactType?: 'vertical' | 'horizontal' | null;
+  allowOverlap?: boolean;
+  preventCollision?: boolean;
+  transformScale?: number;
```
[the remark this answers is not in the document]
Comment (drodil) [reply plugins/home/src/components/CustomHomepage/CustomHomepageGrid.tsx]: Added comments and also made these a bit more clear.

File: .changeset/dull-bottles-teach.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-home': patch
+---
+
+Allow more customization for the CustomHomepageGrid
```
[the remark this answers is not in the document]
Comment (drodil) [reply .changeset/dull-bottles-teach.md:5]: Wow, the storybook is quite cool way to preview these. Added new template for the home. Decided to put it in own file just to make it a bit more clear.

**Comment by drodil** [comment]
Found also small bugs with some of the components getting rendered badly so fixed those in this same PR. Nothing too serious but small improvements for widget rendering.

**Review by jhaals — APPROVED** [review]
Sorry for completely missing this PR. Lets 🚢!

**Pull request merged**